index.js

```diff
@@ -46,244 +42,248 @@ class AntiSpam extends Events.EventEmitter {
 				return false;
 			}
-
-			let msgToSend = formatString(options.kickMessage, msg);
-
-			await msg.channel.send(msgToSend).catch((e) => {
-				if (options.verbose) console.error(e);
-			});
-			return true;
 		};
 
-		const warnUser = async (msg) => {
-			// Mark the user as warned
-			warnedUsers.push(msg.author.id);
+		const warnUser = async () => {
+			data.warnedUsers.push(message.author.id);
 			this.emit('warnAdd', message.member);
 
-			let msgToSend = formatString(options.warnMessage, msg);
-
-			await msg.channel.send(msgToSend).catch((e) => {
-				if (options.verbose) console.error(e);
-			});
+			if (options.warnMessage)
+				await message.channel
+					.send(format(options.warnMessage, message))
+					.catch(e => {
+						if (options.verbose) console.error(e);
+					});
 
 			return true;
 		};
 
-		users.push({
+		data.users.push({
 			time: Date.now(),
-			author: message.author.id
+			id: message.author.id
 		});
 
-		messageCache.push({
+		data.messageCache.push({
 			content: message.content,
 			author: message.author.id
 		});
 
-		let messageMatches = messageCache.filter(
-			(m) => m.content === message.content && m.author === message.author.id
+		const messageMatches = data.messageCache.filter(
+			m => m.content === message.content && m.author === message.author.id
 		).length;
-		let spamMatches = users.filter(
-			(u) =>
-				u.time > Date.now() - options.maxInterval &&
-				u.author === message.author.id
+		const spamMatches = data.users.filter(
+			u =>
+				u.time > Date.now() - options.maxInterval && u.id === message.author.id
 		).length;
 
 		if (
-			!warnedUsers.includes(message.author.id) &&
+			!data.warnedUsers.includes(message.author.id) &&
 			(spamMatches === options.warnThreshold ||
 				messageMatches === options.maxDuplicatesWarning)
 		) {
-			warnUser(message);
-			this.emit('warnEmit', message.member);
+			if (options.warnEnabled) warnUser(message);
+			this.emit(
+				'spamThresholdWarn',
+				message.member,
+				messageMatches === options.maxDuplicatesWarning
+			);
+			return true;
 		}
 
 		if (
-			options.kickEnabled &&
-			!kickedUsers.includes(message.author.id) &&
+			!data.kickedUsers.includes(message.author.id) &&
 			(spamMatches === options.kickThreshold ||
 				messageMatches === options.maxDuplicatesKick)
 		) {
-			await kickUser(message);
-			this.emit('kickEmit', message.member);
+			if (options.kickEnabled) await kickUser(message);
+			this.emit(
+				'spamThresholdKick',
+				message.member,
+				messageMatches === options.maxDuplicatesKick
+			);
+			return true;
 		}
 
 		if (
-			(options.banEnabled && spamMatches === options.banThreshold) ||
+			spamMatches === options.banThreshold ||
 			messageMatches === options.maxDuplicatesBan
 		) {
-			await banUser(message);
-			this.emit('banEmit', message.member);
+			if (options.banEnabled) await banUser(message);
+			this.emit(
+				'spamThresholdBan',
+				message.member,
+				messageMatches === options.maxDuplicatesBan
```

Comment:
I don't see the point of this Boolean

Reply:
the boolean passed into emit? its so that when the event is emitted, you can use that boolean to know if the threshold was met via spamming random messages, or spamming the same, might be useful to someone

Reply:
Yes, but we already know that the threshold was met right? Because if the event is emitted it must have been reached

Reply:
yeah, just you dont know if it was sending the same message over and over or just headbutting the keyboard with random keys or smth

Reply:
I don't see the point in sending this value with the emit. 
It's like this:
```js
let a = true;
let myEmitter = new EventEmitter();
if(a){
    myEmitter.emit("a_is_true", a);
};
myEmitter.on("a_is_true", (aValue) => {
// In this case aValue is useless because we know that its value is inevitably true
}
```
So I think:
```js
this.emit(
    'spamThresholdBan',
    message.member
)
```
would be pretty fine.

Reply:
I'm sorry I didn't read it right. 😅